### DISCUSSION TITLE
Row Select

### DISCUSSION THREAD

**Discussion by davidsaenz2003** [glideapps/glide-data-grid#597] [General]
Can I set a row as selected by default? And is it also possible to prevent clicking anywhere in the grid from unchecking the selected rows?

![image](https://user-images.githubusercontent.com/114322714/211879390-8a42ea30-1af0-4fc2-b61e-975eb70d510e.png)

**Comment by nightire** [comment]
By using `gridSelection` property with `CompactSelection`, you can set the current selection, including selected cells, columns, or rows:

```ts
gridSelection: {
  current: {
    // selection by single cell
    cell: [1, 1],
    // selection by multiple adjacent cells
    range: { height: 2, width: 2, x: 2, y: 2 },
    // more stackable regions
    rangeStack: [{ height: 2, width: 4, x: 1, y: 3 }],
  },
  // selection by column(s)
  columns: CompactSelection.fromSingleSelection(0),
  // selection by row(s)
  rows: CompactSelection.fromSingleSelection([6, 8]),
}

**Reply by davidsaenz2003** [reply]
Great, I'm going to try that method, thank you, hey and another question, is it possible to prevent clicking anywhere in the grid from unchecking the selected rows?

**Reply by jassmith** [reply]
In the `onGridSelectionChange` callback, ensure your row is always selected. `gridSelection` works in much the same way `value` works on an `input` in react. If you set `value` you probably also want to set `onChange` or it wont really ever update. Same applies to `gridSelection` and `onGridSelectionChange`. You should think of them as direct parallels.

**Reply by islittlezhou** [reply]
> In the `onGridSelectionChange` callback, ensure your row is always selected. `gridSelection` works in much the same way `value` works on an `input` in react. If you set `value` you probably also want to set `onChange` or it wont really ever update. Same applies to `gridSelection` and `onGridSelectionChange`. You should think of them as direct parallels.

```
const handleGridSelectionChange = (newSel) => {
    let newRows = CompactSelection.empty();
    if (newSel.current !== undefined) {
      newRows = newRows.add([newSel.current.range.y, newSel.current.range.y + newSel.current.range.height]);
    }
    for (const b of newSel.current?.rangeStack ?? []) {
      newRows = newRows.add([b.y, b.y + b.height]);
    }
    setSelection({
      ...newSel,
      rows: newRows
    });
}
```

The above code can select the row when clicking on a cell, and can select multiple rows when clicking and dragging with the mouse. When pressing the CTRL or SHIFT keys, multiple rows can be selected consecutively. But I have encountered a problem: when I select multiple rows, I hope to cancel the selection status of that row by pressing the CTRL key and left clicking on the selected row with the mouse. I have been thinking for a long time, but have not found a suitable solution. Do you have any good suggestions?
